### PR TITLE
[WIP] test: let's use vite-jest for testing

### PR DESCRIPTION
**Pull request by sodatea**
First, thanks for the great example app!

I was developing [vite-jest](https://github.com/sodatea/vite-jest/blob/main/packages/vite-jest/README.md) and used this repository as a real-world test case.
Now after v0.1.3 all the current tests are passing. So I'm opening this PR in case you are interested in joining the experiment of this new vite-jest integration.

Main changes in this PR:
- removed `babel-jest`, `ts-jest` in favor of `vite-jest`
- a few refactors as all test specs are native ES modules now:
	- `jest` object is imported from `@jest/globals`
	- use `jest.unstable_mockModule` instead of `jest.mock`
- added 3 `@ts-expect-error` comments to pass the git pre-commit hook. These type checkings seem to have been failing before my refactor.

The coverage report is not working at the moment, though.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,26 +62,8 @@
...
-    "preset": "ts-jest",
-    "globals": {
-      "ts-jest": {}
-    },
-    "testEnvironment": "jsdom",
-    "transform": {
-      "^.+\\.vue$": "vue3-jest",
-      "^.+\\js$": "babel-jest"
-    },
-    "collectCoverageFrom": [
```
Comment (mutoe) [inline package.json:75]: Can we continue to maintain coverage requirements?

**Comment by mutoe** [comment]
@sodatea  Sounds great, bro. I don't know if you represent the official Vite, but before it is officially released (at least the beta), I think this can only be pending, or merged into a separate branch for tracking :)

File: src/components/ArticleDetailComments.spec.ts [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
-import { getCommentsByArticle } from 'src/services/comment/getComments'
-import { deleteComment } from 'src/services/comment/postComment'
 import asyncComponentWrapper from 'src/utils/test/async-component-wrapper'
 import fixtures from 'src/utils/test/fixtures'
-import ArticleDetailComments from './ArticleDetailComments.vue'
 
-jest.mock('src/services/comment/getComments')
-jest.mock('src/services/comment/postComment')
+jest.unstable_mockModule('src/services/comment/getComments', () => ({
+  getCommentsByArticle: jest.fn(),
```
Comment (mutoe) [inline src/components/ArticleDetailComments.spec.ts:9]: Whether I still need to declare the mocked function here? Is it necessary?

**Comment by sodatea** [comment]
> but before it is officially released (at least the beta), I think this can only be pending, or merged into a separate branch for tracking :)

Yeah, it makes sense. I'll ping you again when the package reaches beta.

File: src/components/ArticleDetailComments.spec.ts (hunk 2 above)
Comment (sodatea) [reply src/components/ArticleDetailComments.spec.ts:9]: Yes, it is necessary. The jest mock API for ES modules only accepts a factory function for now, so the exports must be mocked and returned from that factory function.

**Comment by mutoe** [comment]
Hi @sodatea , after the release of Vitest version 1.0, this project has been migrated to Vitest. Also this PR has been too long, close it for now. Welcome to open it again if necessary :D

**Pull request closed without merging**